### PR TITLE
Filter on 5xx status code to retry only

### PR DESCRIPTION
**Pull request by xiazcy**
Regex string parsing for 5xx status code, as AmazonClientExceptions do not return status codes. 

List of S3 upload error codes: https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList

### CODE REVIEW

File: neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java [hunk 1]
```diff
@@ -348,10 +350,13 @@ private void uploadExportFilesToS3(TransferManager transferManager, File directo
 
                 AmazonClientException amazonClientException = upload.waitForException();
 
+                // override AmazonClientException's isRetryable instead?
                 if (amazonClientException != null){
                     String errorMessage = amazonClientException.getMessage();
+                    // only retry on 5xx? or no-retry on 4xx?
+                    Matcher exMsgStatusCodeMatcher = Pattern.compile("Status Code: (5\\d+)").matcher(errorMessage);
```
Comment (vkagamlyk) [inline neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java:357]: better to make static variable for compiled pattern once and reuse

Comment (xiazcy) [reply neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java]: Updated

File: neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java [hunk 2]
```diff
@@ -104,6 +106,7 @@ public static ObjectTagging createObjectTags(Collection<String> profiles) {
     private final Collection<String> profiles;
     private final Collection<CompletionFileWriter> completionFileWriters;
     private final AtomicReference<S3ObjectInfo> result = new AtomicReference<>();
+    private static final Pattern STATUS_CODE_5XX_PATTERN = Pattern.compile("Status Code: (5\\d+)");
```
Comment (cole-bq) [inline neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java:109]: Is it always trying to match 3 digit 5XX status codes? Could change regex to `"Status Code: (5\\d\\d)"` to be a bit more specific.

**Review by vkagamlyk — APPROVED** [review]
LGTM

File: neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java (hunk 2 above)
Comment (xiazcy) [reply neptune-export/src/main/java/com/amazonaws/services/neptune/export/ExportToS3NeptuneExportEventHandler.java:109]: There should only be 3 digit 5xx status code, so I think either works?

**Pull request merged**